drivers/storage/portworx/util/util.go

```diff
@@ -1402,30 +1402,12 @@ func CountStorageNodes(
 		return -1, fmt.Errorf("failed to enumerate nodes: %v", err)
 	}
 
-	k8sNodeList := &v1.NodeList{}
-	err = k8sClient.List(context.TODO(), k8sNodeList)
-	if err != nil {
-		return -1, err
-	}
-	k8sNodesStoragePodCouldRun := make(map[string]bool)
-	for _, node := range k8sNodeList.Items {
-		shouldRun, shouldContinueRunning, err := k8sutil.CheckPredicatesForStoragePod(&node, cluster, nil)
-		if err != nil {
-			return -1, err
-		}
-		if shouldRun || shouldContinueRunning {
-			k8sNodesStoragePodCouldRun[node.Name] = true
-		}
-	}
-
 	// Use the quorum member flag from the node enumerate response if all the nodes are upgraded to the
 	// newer version. The Enumerate response could be coming from any node and we want to make sure that
 	// we are not talking to an old node when enumerating.
 	useQuorumFlag := true
 	for _, node := range nodeEnumerateResponse.Nodes {
-		if node.Status == api.Status_STATUS_DECOMMISSION {
```

Comment:
We should still check that the node is not decommissioned when figuring out whether to use the non_quorum_member flag or not. For some reason, if the decommissioned node is still part of the cluster and is the only node on an older version, then we should still go ahead and use the quorum flag for counting storage nodes.

Without this check we will end up setting useQuorumFlag = false.